### ISSUE TITLE
questions about redisson 

### ISSUE THREAD

**Issue by gmailisgone** [redisson/redisson#1597]
These are questions about redissons rather than issues. 

I plan to use redisson in our company project, Use RLocalCachedMap to cache some business orders, but still have some questions about RLocalCachedMap since I can't get them from Wiki, API, Issue discussion in GitHub. 

1. read from map of RLocalCachedMap will always return the NEWEST data in real-time. 

2. clearLocalCache just clear the local data in JVM, it will not clear the data in Redis server, next read of clearLocalCache will sync the data from redis server to local JVM, right?

3. How to get a Pro version of redisson, please send a email to haiyang.pan@gmail.com

**Comment by mrniko** [comment]
> 1. read from map of RLocalCachedMap will always return the NEWEST data in real-time.

Yes. If you want transactional updates use `RTransaction.getLocalCachedMap`

> 2. clearLocalCache just clear the local data in JVM, it will not clear the data in Redis server, next read of clearLocalCache will sync the data from redis server to local JVM, right?

Right, read operation will load map entry to local cache.

> 3. How to get a Pro version of redisson, please send a email to haiyang.pan@gmail.com

Please check your mail

**Comment by gmailisgone** [comment]
@mrniko 
thanks for your reply. 

for now, I am using  getReadWriteLock to avoid the concurrency update, I think it should be OK.

I have received the PRO key, but there is no price in it.  I have to report it to my company, even community version is enough for earlier stage use. 

I think redission is a good product, but I suggest you to make more detail documents about the APIs/Designs/Examples in order to let developers to understand it more faster, people for now are impatient and don't want to spend too much time on new things.
